### PR TITLE
[GOBBLIN-1359] Fix the error that IOException thrown by writer be swallowed by Reactivex

### PR DESCRIPTION
**Pull request by ZihanLi58**
…llowed by Reactivex

Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

### JIRA
- [ ] My PR addresses the following [Gobblin JIRA](https://issues.apache.org/jira/browse/GOBBLIN/) issues and references them in the PR title. For example, "[GOBBLIN-XXX] My Gobblin PR"
    - https://issues.apache.org/jira/browse/GOBBLIN-1359

### Description
- [ ] Here are some details about my PR, including screenshots (if applicable):
Fix the bug in Reactivex code which swallows the exception and leaves the task hanging

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Create dummy IOException in writer and make sure it can cause the task failure

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by ZihanLi58** [comment]
@sv2000 Can you help take a look at this minor change for the Reactivex bug

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/fork/Fork.java [hunk 1]
```diff
@@ -227,7 +227,10 @@ public void consumeRecordStream(RecordStreamWithMetadata<D, S> stream)
 
           r.ack();
         }
-      }, e -> logger.error("Failed to process record.", e),
+      }, e -> {
+          logger.error("Failed to process record.", e);
+          throw(new RuntimeException(e));
```
Comment (sv2000) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/fork/Fork.java:232]: Any reason why it is being wrapped into RuntimeException? Can we just rethrow e as is?

**Review by sv2000 — CHANGES_REQUESTED** [review]
A minor comment.

File: gobblin-runtime/src/main/java/org/apache/gobblin/runtime/fork/Fork.java [hunk 2]
```diff
@@ -227,7 +227,10 @@ public void consumeRecordStream(RecordStreamWithMetadata<D, S> stream)
 
           r.ack();
         }
-      }, e -> logger.error("Failed to process record.", e),
+      }, e -> {
+          logger.error("Failed to process record.", e);
+          throw((Exception) e);
```
Comment (sv2000) [inline gobblin-runtime/src/main/java/org/apache/gobblin/runtime/fork/Fork.java:232]: Sorry - I did not realize earlier that e is a throwable, in which case I now understand why you had to wrap it into a RuntimeException. I think that would be better than casting to an Exception.

**Review by sv2000 — APPROVED** [review]
+1. LGTM.

**Pull request closed without merging**